### DISCUSSION TITLE
Manual Vague

### DISCUSSION THREAD

**Discussion by smartbyte** [dzungpv/mitsubishi2MQTT#45] [General]
Dear Developer,

I am getting multiple compile and now ld-linking errors when compiling the project. Either with Arduino or with IDF.

Now I wanted to flash the prebuild flash, but I don't know how to do that. I tried directly with the esptool, but this does not work.

Can I please ask you to elaborate on the following entry in the README:
"Step 1: Flash the sketch with flash size include SPIFFS option."

I tried to flash the binary with this command (flashing works!) butt then on the ESP32 it does not work:
esptool.py -p /dev/ttyUSB0 -b 460800 --before default_reset --after hard_reset --chip esp32  write_flash --flash_mode dio --flash_size detect --flash_freq 40m 0x0000 mitsubishi2MQTT_ESP32DEV_IDF.bin

My hardware is a ESP32-WROOM-32 module (Is that incompatible with above image?)

**Comment by dzungpv** [comment]
Because with the esp32 you need additional boot-loader, partition and init ota file, the flash command also complex too. You can follow this https://github.com/dzungpv/mitsubishi2MQTT/issues/31#issuecomment-2230570049 to build and flash yourself

**Comment by dzungpv** [comment]
> Dear Developer,
> 
> I am getting multiple compile and now ld-linking errors when compiling the project. Either with Arduino or with IDF.
> 
> Now I wanted to flash the prebuild flash, but I don't know how to do that. I tried directly with the esptool, but this does not work.
> 
> Can I please ask you to elaborate on the following entry in the README: "Step 1: Flash the sketch with flash size include SPIFFS option."
> 
> I tried to flash the binary with this command (flashing works!) butt then on the ESP32 it does not work: esptool.py -p /dev/ttyUSB0 -b 460800 --before default_reset --after hard_reset --chip esp32 write_flash --flash_mode dio --flash_size detect --flash_freq 40m 0x0000 mitsubishi2MQTT_ESP32DEV_IDF.bin
> 
> My hardware is a ESP32-WROOM-32 module (Is that incompatible with above image?)

I have add all-in-one bin file, you can follow guide here and flash https://github.com/dzungpv/mitsubishi2MQTT?tab=readme-ov-file#how-to-flash-pre-build-bin-file-for-esp32-only
